pkg/component/delete/delete_test.go

```diff
@@ -0,0 +1,461 @@
+
+					selector := componentlabels.GetSelector(componentName, "app")
+					client.EXPECT().GetOnePodFromSelector(selector).Return(nil, &kclient.PodNotFoundError{Selector: selector})
+
+					labels := componentlabels.GetLabels(componentName, "app", false)
+					client.EXPECT().Delete(labels, false).Return(nil)
+					return client
+				},
+			},
+			args: args{
+				devfileObj:    odoTestingUtil.GetTestDevfileObj(fs),
+				componentName: componentName,
+			},
+			wantErr: false,
+		},
+		{
+			name: "component deleted successfully even though user is forbidden",
+			fields: fields{
+				kubeClient: func(ctrl *gomock.Controller, componentName string) kclient.ClientInterface {
+					client := kclient.NewMockClientInterface(ctrl)
+
+					selector := componentlabels.GetSelector(componentName, "app")
+					client.EXPECT().GetOnePodFromSelector(selector).Return(nil, kerrors.NewForbidden(schema.GroupResource{}, "", nil))
+
+					labels := componentlabels.GetLabels(componentName, "app", false)
+					client.EXPECT().Delete(labels, false).Return(nil)
+					return client
+				},
+			},
+			args: args{
+				devfileObj:    odoTestingUtil.GetTestDevfileObj(fs),
+				componentName: componentName,
+			},
+			wantErr: false,
+		},
+		{
+			name: "failed to delete the component because of an un-ignorable error",
+			fields: fields{
+				kubeClient: func(ctrl *gomock.Controller, componentName string) kclient.ClientInterface {
+					client := kclient.NewMockClientInterface(ctrl)
+					selector := componentlabels.GetSelector(componentName, "app")
+					client.EXPECT().GetOnePodFromSelector(selector).Return(nil, errors.New("some un-ignorable error"))
+					return client
+				},
+			},
+			args: args{
+				devfileObj:    odoTestingUtil.GetTestDevfileObj(fs),
+				componentName: componentName,
+			},
+			wantErr: true,
+		},
+		{
+			name: "component deleted successfully with a devfile that has PreStopEvents",
+			fields: fields{
+				kubeClient: func(ctrl *gomock.Controller, componentName string) kclient.ClientInterface {
+					client := kclient.NewMockClientInterface(ctrl)
+					selector := componentlabels.GetSelector(componentName, "app")
+					client.EXPECT().GetOnePodFromSelector(selector).Return(odoTestingUtil.CreateFakePod(componentName, "runtime"), nil)
+					cmd := []string{"/bin/sh", "-c", "cd /projects/nodejs-starter && echo \"Hello World!\""}
+					client.EXPECT().ExecCMDInContainer("runtime", "runtime", cmd, gomock.Any(), gomock.Any(), nil, false).Return(nil)
+					labels := componentlabels.GetLabels(componentName, "app", false)
+					client.EXPECT().Delete(labels, false).Return(nil)
+					return client
+				},
+			},
+			args: args{
+				devfileObj:    odoTestingUtil.GetTestDevfileObjWithPreStopEvents(fs, "runtime", "echo \"Hello World!\""),
+				componentName: componentName,
+			},
+			wantErr: false,
+		},
+		{
+			name: "failed to delete component where a devfile that has PreStopEvents because pod is not in the running state",
```

Comment:
Delete should happen even if the event fails. Otherwise, it would be easy to get into a situation where it is impossible to delete the component due to a bug/type/mistake in PreStop event. 

If this turns out to be a problem, we can consider changing this in the future, to not delete if it fails, and add flag `--ignore-even-failures`.
To keep things simple for now, lets just ignore prestop events failures by default .